lib/network/modules/InteractionHandler.js

```diff
@@ -413,19 +436,40 @@ class InteractionHandler {
    */
   onDragEnd(event) {
     this.drag.dragging = false;
-    let selection = this.drag.selection;
-    if (selection && selection.length) {
-      selection.forEach(function (s) {
-        // restore original xFixed and yFixed
-        s.node.options.fixed.x = s.xFixed;
-        s.node.options.fixed.y = s.yFixed;
-      });
-      this.selectionHandler._generateClickEvent('dragEnd', event, this.getPointer(event.center));
-      this.body.emitter.emit('startSimulation');
-    }
-    else {
+    
+    if (this.body.selectionBox.show) {
+      this.body.selectionBox.show = false;
+      const selectionBoxPosition = this.body.selectionBox.position 
+      const selectionBoxPositionMinMax = {
+        minX: Math.min(selectionBoxPosition.start.x, selectionBoxPosition.end.x),
+        minY: Math.min(selectionBoxPosition.start.y, selectionBoxPosition.end.y),
+        maxX: Math.max(selectionBoxPosition.start.x, selectionBoxPosition.end.x),
+        maxY: Math.max(selectionBoxPosition.start.y, selectionBoxPosition.end.y),
+      }
+
+      const toBeSelectedNodes = Object.values(this.body.nodes).filter((node) => (
```

Comment:
Shouldn't you use `this.body.nodeIndices`? Because `Object.values(this.body.nodes)` includes hidden and support nodes.

Reply:
You're right.
I've fixed it to your suggestion so the selection does not include hidden and support nodes.